### PR TITLE
feat: Add tag read, write, and delete commands

### PR DESCRIPTION
**Pull request by bhavanki**
The new commands are for reading, writing, and deleting tags on the
underlying resources representing secrets.

For now, the commands are only implemented for SSM parameters.

### CODE REVIEW

File: store/ssmstore.go [hunk 1]
```diff
@@ -126,6 +127,72 @@ func (s *SSMStore) Read(ctx context.Context, id SecretId, version int) (Secret,
 	return s.readVersion(ctx, id, version)
 }
 
+func (s *SSMStore) WriteTags(ctx context.Context, id SecretId, tags map[string]string, deleteOtherTags bool) error {
+	// list the current tags
+	currentTags, err := s.ReadTags(ctx, id)
```
Comment (alecjacobs5401) [inline store/ssmstore.go:132]: we only use the current tags if we're doing an idempotent tag update and deleting other tags, right?

If so, can we only ready the tags when we're going to do something with those current tags?

File: cmd/tag-write.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	tagWriteCmd = &cobra.Command{
+		Use:   "write <service> <key> <tag>...",
+		Short: "Write tags for a specific secret",
+		Args:  cobra.MinimumNArgs(3),
+		RunE:  tagWrite,
+	}
+)
+
+func init() {
+	tagWriteCmd.Flags().BoolVar(&deleteOtherTags, "delete-other-tags", false, "Delete tags not specified in the command")
```
Comment (alecjacobs5401) [inline cmd/tag-write.go:28]: This is a descriptive variable name, but I'm curious if there is precedent for CLIs to have an argument that is semantically the same but more conventional? Maybe `--reset` or `--sync` or `--prune`? Really not asking for a change just opening up a discussion about what a conventional flag someone might expect to exist on a CLI that has this behavior in a low surprise manner

**Review by alecjacobs5401 — COMMENTED** [review]
Few open questions - nothing really jumps out at me immediately here otherwise

File: cmd/tag-write.go (hunk 2 above)
Comment (bhavanki) [reply cmd/tag-write.go:28]: I didn't really do a survey, but I did ask OpenAI for suggestions and chose this 😆

File: store/ssmstore.go (hunk 1 above)
Comment (bhavanki) [reply store/ssmstore.go]: Good catch! Fixed.

File: cmd/tag-write.go (hunk 2 above)
Comment (alecjacobs5401) [reply cmd/tag-write.go:28]: Hah! Funny. I asked open AI the same thing and it suggested those (specifically sync and prune) 😂

**Review by alecjacobs5401 — APPROVED** [review]

File: cmd/tag-write.go (hunk 2 above)
Comment (bhavanki) [reply cmd/tag-write.go:28]: I guess it depends on the question phrasing and OpenAI's mood that day!

**Pull request merged**